rfcs/0024-python-scope-package-set.md

```diff
@@ -0,0 +1,145 @@
+- maintainer: a maintainer of an expression as typically noted in the `meta.maintainers` field
+- package set maintainer: a maintainer of the Python package set as noted in the `CODEOWNERS` file
+
+Examples on the usage of application and library:
+- `scipy` is a collection of routines for scientific computing. It is used for development purposes or as a library and is therefore considered a *library*
+- `calibre` is a program for managing e-books that happens to be written in Python. The package is therefore considered an *application*
+- `pytest` is a test runner. While a program, it can only be used in conjunction with the Python environment where it needs to import the modules that are to be tested. It is therefore considered a *library*
+
+## Scope of the package set
+
+The purpose of the package set is to provide a set of compatible libraries that
+can be used for development purposes and as dependencies by Python applications.
+Applications are not considered part of the package set and their expressions
+shall thus also be placed elsewhere.
+
+For each NixOS release, a set of compatible libraries shall be formed
+that functions on all supported platforms, interpreter versions and
+implementations.
+
+That in effect means only one version of a package is permitted as having
+multiple versions in the set may lead to collisions. Exceptions can be made,
+like in the case of build or test-time dependencies such as the `pytest` test runner.
+
+The definitions from the previous section shall be used to decide whether an
+expression should be part of the Python package set or not. The following
+special cases exist.
+
+### Package that is used as a both an application and a library
+In certain cases a package is used as both an application and a library. The
+predominant use case determines whether the expression is considered an
+application or a library. If both use cases are considered common, then the
+expression shall be part of the package set and a special alias shall we added
+to the top-level attribute set. This special attribute will mark the package as an
+application and shall be documented in the Nixpkgs manual.
+
+### Application-specific dependencies
+A special case to consider are application-specific dependencies. These are
+dependencies that have been factored out into separate packages but are only
+intended to be used by the parent application. Examples of such packages are
+`buildbot` plugins or the `sabyenc` module for `sabnzbd`. As such packages are
+of no use in a development environment, they should be bundled with the
+expression of the parent application.
+
+## Contributing and maintaining expressions
+
+Maintenance of the expressions takes effort. All expressions shall therefore
+have a maintainer. Packages that have no maintainer may be removed from the
+package set or marked as broken in case they seem to be unused, untouched in an
+extended period of time, or in whatever way cause additional work for other
```

Comment:
What if there are no upstream updates and the package actually works well without them anyway?

Reply:
I surely agree with that. Maybe put the emphasis on this effort? «Packages that have no maintainer may be removed or marked as broken as soon as they require any additional maintenance effort, or there are any indications of dangerous bugs or vulnerabilities.»